lib/chat/chunked_files_multikey.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Chat.ChunkedFilesMultikey do
+  @moduledoc false
+
+  alias Chat.Db
+  alias Chat.Utils
+
+  @chunk_group_size 1_000_000_000
+  @initial_offset_range 0..1000
+
+  def generate(file_key, file_size, <<initial_secret::binary-size(32)>>) do
+    secrets_needed_amount = trunc(file_size / @chunk_group_size)
+
+    secrets_needed =
+      for amount <- 0..secrets_needed_amount,
+          amount > 0,
+          do: encrypt_secret(additional_secret(), initial_secret)
+
+    Db.put({:file_secrets, file_key}, secrets_needed |> Enum.join(""))
+  end
+
+  def get_secret(_file_key, offset, <<initial_secret::binary-size(32)>>)
+      when offset in @initial_offset_range,
+      do: initial_secret
+
+  def get_secret(file_key, offset, _initial_secret) do
+    secrets = Db.get({:file_secrets, file_key})
+    secrets = for <<chunk::binary-size(32) <- secrets>>, do: <<chunk::binary-size(32)>>
```

Comment:
building a list to get only value  looks inefficient
use something like `String.slice` but for bitstrings